components/search_engines/search_engines_pref_names.h

```diff
@@ -0,0 +1,12 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
```

Comment:
I thought about extending chromium file via `chromium_src/components/search_engines/search_engines_pref_names.h`.
That way you don't need to add `brave_components_search_engines_sources` in `.gn`.